library/systemd/src/lib/yast2/service_configuration.rb

```diff
@@ -0,0 +1,194 @@
+require "yast"
+
+Yast.import "SystemdService"
+
+module Yast2
+  # Class that holds configuration for single or multiple services. It allows
+  # to read current status from system. It allows to modify target state. And it
+  # allows to write that target state so system reflects it.
+  # Its common usage is with {Yast2::ServiceWidget} which allows to modify
+  # configuration and write it when user approve whole dialog.
+  class ServiceConfiguration
+    include Yast::Logger
+
+    # services managed by this configuration
+    attr_reader :services
+
+    # creates new configuration that holds state for given services
+    # @param services<Yast::SystemdServiceClass::Service> services to configure
+    # @param reload<true,false> if use reload instead of restart action. If
+    #   service does not support reload or does not run, then restart is used.
+    #
+    # @example three services
+    #   config = Yast2::ServiceConfiguration.new(s1, s2, s3)
+    def initialize(*services)
+      if services.any? { |s| !s.is_a?(Yast::SystemdServiceClass::Service) }
+        raise ArgumentError, "Services can be only Systemd Service - #{services.inspect}"
+      end
+
+      @services = services
+    end
+
+    # reads system status of services. Can be also used to reread current status.
+    # @raise <Yast::SystemctlError> when read services state failed
+    def read
+      # safe initial state when exception happen
+      @status = :unknown
```

Comment:
Why not moving this initialization to the `initialize` method?

Reply:
because I though that for read it can take some time. So it is more consistent with common yast workflow when read and write have own progress. So you can inform about reading services and also showing progress for it.

Reply:
I guess Imo means to only move the assignments, not the read* calls.

Reply:
But moving it to `initialize` makes more difficult to check whether `read` has been already called.

Reply:
Ok, that's a valid point (you want to know whether `read` was called or not).